src/main/java/com/algorand/algosdk/templates/DynamicFee.java

```diff
@@ -152,13 +152,13 @@ public static SignedDynamicFee SignDynamicFee(final ContractTemplate contract, f
         txn2.setLease(new Lease(txn.lease));
         Account.setFeeByFeePerByte(txn2, BigInteger.valueOf(feePerByte));
 
-        TxGroup.assignGroupID(txn, txn2);
+        TxGroup.assignGroupID(txn2, txn);
```

Comment:
It might help readability to name the `txn`s here, like `mainPay` and `feeReimburse` or `mainTxn` and `feeTxn` or something, rather than numbered txns.